gen/AArch64Compile_gen.ml

```diff
@@ -2762,11 +2762,22 @@ module Make(Cfg:Config) : XXXCompile_gen.S =
         let n2 = C.find_non_pseudo n in
         let loc = match n2.C.evt.C.loc with
         | Data loc -> loc
-        | Code _ -> Warn.fatal "Can't create address dependency for a code location" in
+        | Code lab -> match f with
+          | CMO (_,Next) -> lab
+          | _ -> Warn.fatal "Address dependency to code location must be directly followed by DC.CVAUn or IC.IVAUn" in
+        (try ignore (C.find_node
+          (fun m -> match m.C.edge.E.edge with
```

Comment:
Could you do?
```suggestion
        (try ignore (C.find_edge
          (function
```
Another suggestion would be to add an `iter_node/iter_edge` function in cycle.ml which would simplify the code here as well.

Reply:
i dont think we can use find_edge, because we need to get the edge type from the node, which contains the actual edge and two atom options, and then edge. find_edge would only work for the actual edge. We could change the input to the function to be n.C.edge, but i'm not sure that would make the code much cleaner.

i agree an iter_node/iter_edge function would be very useful. Would it be ok to do it in a different PR? just because there are other places it could also be used, and i think it could use its own PR because of that